src/domain/usecases/ListAllUsernamesUseCase.ts

```diff
@@ -0,0 +1,11 @@
+import { UseCase } from "../../CompositionRoot";
```

Comment:
A use case should never return primitive types, this is a code smell that we saw in the coding dojo

In Clean Architecture, a use case should always return entities.

The presentation layer should map user list to get userNames if required.

In this case the entity is UserLogic (can we change to User and User to UserProps?)